tests/lib/test_openstack_flavor.py

```diff
@@ -0,0 +1,211 @@
+            name=mocked_flavor_data.name,
+            ram=mocked_flavor_data.ram,
+            vcpus=mocked_flavor_data.vcpus,
+            disk=mocked_flavor_data.disk,
+            flavorid="auto",
+            ephemeral=mocked_flavor_data.ephemeral,
+            swap=mocked_flavor_data.swap,
+            rxtx_factor=mocked_flavor_data.rxtx_factor,
+            is_public=mocked_flavor_data.is_public,
+        )
+
+    def test_empty_dict(self):
+        """
+        Test that method set_flavor_specs not called when empty_dictionary passed
+        """
+        empty_flavor_spec = {}
+        mocked_cloud_name = NonCallableMock()
+        flavor_id = NonCallableMock()
+        self.instance.set_flavor_specs(mocked_cloud_name, flavor_id, empty_flavor_spec)
+        self.api.set_flavor_specs.assert_not_called()
+
+    def test_flavor_id_get_specs(self):
+        """
+        Test getting the extra specs from a flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        mocked_flavor_id = NonCallableMock()
+
+        flavor = self.instance.get_flavor(
+            cloud_account=mocked_cloud_name, flavor_name=mocked_flavor_id
+        )
+
+        assert flavor == self.api.get_flavor.return_value
+        self.api.get_flavor.assert_called_once_with(mocked_flavor_id)
+
+    def test_flavor_object_get_specs(self):
+        """
+        Test getting the extra specs from a flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        mocked_flavor = NonCallableMock()
+
+        extra_specs = Mock()
+        self.instance.get_flavor_specs = Mock()
+
+        self.instance.get_flavor_specs.return_value = extra_specs
+
+        self.instance.get_flavor_specs(mocked_cloud_name, mocked_flavor)
+
+        self.instance.get_flavor_specs.assert_called_once_with(
+            mocked_cloud_name, mocked_flavor
+        )
+
+    def test_set_specs(self):
+        """
+        Test setting the extra specs for a flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        flavor_id = NonCallableMock()
+        extra_specs = NonCallableMock()
+
+        self.instance.set_flavor_specs(mocked_cloud_name, flavor_id, extra_specs)
+        self.mocked_connection.assert_called_once_with(mocked_cloud_name)
+        self.api.set_flavor_specs.assert_called_once_with(flavor_id, extra_specs)
+
+    def test_empty_list_migrate_flavors(self):
+        # mock params
+        mocked_source_cloud = NonCallableMock()
+        mocked_dest_cloud = NonCallableMock()
+        # mock return value
+        missing_flavors = NonCallableMock()
+        # mock functions
+        self.instance.get_missing_flavors = Mock()
+        self.instance.get_flavor = Mock()
+        # prepare return value
+        self.instance.get_missing_flavors.return_value = missing_flavors
+        # call method
+        self.instance.get_missing_flavors(mocked_source_cloud, mocked_dest_cloud)
+        # assert at least one function that is called by method
```

Comment:
We could trim out these comments, since we use the same pattern throughout the code base.
If you'd like to preserve them, maybe a developer section in the README would be a more convenient place to put an example?